src/app/product.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { HttpClient } from '@angular/common/http';
+
+import { Product } from './product';
+import { SoldProducts } from './product';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProductService {
+  private productsUrl = 'https://private-anon-cfb1043af1-swoproducts.apiary-mock.com/soldProducts'; 
+
+  constructor(private http: HttpClient) { }
```

Comment:
![ScreenShot 2021-02-6 o 11 49 56](https://user-images.githubusercontent.com/6479741/107116115-8791bb00-6871-11eb-93b2-62799ed6dfdc.png)
With this design you call API 3 times :) and you should only one. Always review the network tab when you finnish.

Reply:
Added cached Observable to reduce the api requests.